src/test/kotlin/org/rust/ide/actions/RustStringLiteralJoinLinesHandlerTest.kt

```diff
@@ -0,0 +1,102 @@
+package org.rust.ide.actions
+
+import com.intellij.openapi.actionSystem.IdeActions
+import org.intellij.lang.annotations.Language
+import org.rust.lang.RustTestCaseBase
+
+class RustStringLiteralJoinLinesHandlerTest : RustTestCaseBase() {
+    override val dataPath: String = ""
+
+    private fun doTest(
+        @Language("Rust") before: String,
+        @Language("Rust") after: String
+    ) {
+        checkByText("foo.rs", before, after) {
+            myFixture.performEditorAction(IdeActions.ACTION_EDITOR_JOIN_LINES)
+        }
+    }
+
+    fun testNoEscape() = doTest(
+        """
+        fn main() {
+            "Hello<caret>,
+             World"
+        }
+        """,
+        """
+        fn main() {
+            "Hello,<caret> World"
```

Comment:
It does not matter here, but you can use `/*caret*/` now: https://github.com/intellij-rust/intellij-rust/blob/3dd1a3f922f8f59ccfe6b2705781cfb82f2784d4/src/test/kotlin/org/rust/lang/RustTestCaseBase.kt#L91